File: src/main/java/com/sequenceiq/cloudbreak/controller/UserController.java
package com.sequenceiq.cloudbreak.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import com.sequenceiq.cloudbreak.controller.json.UserJson;
import com.sequenceiq.cloudbreak.converter.UserConverter;
import com.sequenceiq.cloudbreak.domain.User;
import com.sequenceiq.cloudbreak.repository.UserRepository;
import com.sequenceiq.cloudbreak.security.CurrentUser;

@Controller
public class UserController {

    @Autowired
    private UserConverter userConverter;

    @Autowired
    private UserRepository userRepository;

    @RequestMapping(method = RequestMethod.GET, value = "/me")
    @ResponseBody
    public ResponseEntity<UserJson> generate(@CurrentUser User user) throws Exception {
        UserJson json = userConverter.convert(userRepository.findOneWithLists(user.getId()));
        return new ResponseEntity<>(json, HttpStatus.OK);
    }

}

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/UserController.kt
package com.sequenceiq.cloudbreak.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.api.endpoint.UserEndpoint
import com.sequenceiq.cloudbreak.domain.CbUser
import com.sequenceiq.cloudbreak.api.model.UserRequest
import com.sequenceiq.cloudbreak.service.user.UserDetailsService

@Component
class UserController : UserEndpoint {

    @Autowired
    private val userDetailsService: UserDetailsService? = null

    @Autowired
    private val authenticatedUserService: AuthenticatedUserService? = null

    override fun evictUserDetails(id: String, userRequest: UserRequest): String {
        userDetailsService!!.evictUserDetails(id, userRequest.username)
        return userRequest.username
    }

    override fun hasResources(id: String): Boolean? {
        val user = authenticatedUserService!!.cbUser
        val hasResources = userDetailsService!!.hasResources(user, id)
        return hasResources
    }

}
